app/queries/consumer_apps/find_or_create_by_query.rb

```diff
@@ -11,16 +11,24 @@ def initialize(app_bundle:, platform:)
 
     def call
       # This guard clause handles the Forem app special case
-      return forem_app_target if @app_bundle == ConsumerApp::FOREM_BUNDLE
+      return forem_consumer_app if @app_bundle == ConsumerApp::FOREM_BUNDLE
 
       # All other ConsumerApp are simply fetched as usual
-      ConsumerApp.find_by(app_bundle: @app_bundle, platform: @platform)
+      relation.find_by(app_bundle: @app_bundle)
     end
 
     private
 
-    def forem_app_target
-      ConsumerApp.create_or_find_by(app_bundle: ConsumerApp::FOREM_BUNDLE, platform: @platform)
+    attr_reader :platform
+
+    def relation
+      # taking advantage of enum scopes in Rails
+      # see https://api.rubyonrails.org/classes/ActiveRecord/Enum.html
+      ConsumerApp.public_send(platform)
```

Comment:
Enums create scopes for the various values. So now we have `ConsumerApp.android`, `ConsumerApp.ios`, this is just a trick to call either of them

Reply:
I had suggested something like this on the original PR and there was a bit of resistance to it. It was exactly this case, but figured I should mention it.